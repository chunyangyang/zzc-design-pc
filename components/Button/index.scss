@import '../style/base.scss';
$cls: '.zds-button';
$cls-group: #{$cls}-group;

//按钮active
@mixin btnActive($color: #fff, $borderColor: #fff, $type: 'default') {
    &#{$cls}-active-#{$type} {
        background: $color;
        border-color: $borderColor;
    }
}

@mixin btnSubStyle() {
    &#{$cls}-sub {
        background-color: $button_sub_background_color;
        color: $button_sub_text_color;
        border: 1px solid $button_sub_border_color;
        @include btnActive($button_sub_active_background_color, $button_sub_border_color, 'sub');

        &#{$cls}-disabled {
            opacity: 0.3;
            background-color: $button_sub_disabled_background_color;
            color: $button_sub_disabled_text_color;
            border: 1px solid $button_sub_disabled_border_color;
        }
    }
}

@mixin btnSpecialStyle() {
    &#{$cls}-special {
        background-color: $button_special_background_color;
        color: $button_special_text_color;
        border: 1px solid $button_special_border_color;
        @include btnActive($button_special_background_color, $button_special_active_border_color, 'special');
    }
}

//按钮基本样式
@mixin btnBase() {
    cursor: pointer;
    font-size: $fs_h1;
    font-weight: 500;
    padding: 11.5px 0;
    line-height: 26px;

    background-color: $button_background_main;
    color: $button_text_color;
    text-align: center;
    border: 1px solid $button_border_main;

    @include flex();
    border-radius: 4px;
    box-sizing: border-box;

    white-space: nowrap;
    vertical-align: middle;
    -ms-touch-action: manipulation;
    touch-action: manipulation;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;

    // 点击反馈
    @include btnActive($buton_main_background_active, $buton_main_border_active, 'default');
    // 次级按钮样式
    @include btnSubStyle();
    // 特殊按钮样式
    @include btnSpecialStyle();

    &#{$cls}-inactive {
        opacity: 0.3;
        background-color: $button_background_inactive;
    }

    &#{$cls}-disabled {
        opacity: 0.2;
        background-color: $button_background_disabled;
        border: 1px solid $button_border_disabled;
    }
}

@mixin btnSize() {
    font-size: $fs_h2;    
    font-weight: 500;
    padding: 7.5px 34.5px;
    line-height: 20px;

    &#{$cls}-middle {
        font-size: $fs_text;
        height: 30px;
        font-weight: 500;
        padding: 4.5px 18px;
        line-height: 20px;
    }

    &#{$cls}-small {
        font-size: $fs_text;        
        font-weight: 500;
        padding: 1px 18px;
        line-height: 20px;
    }
}

// 行内按钮组件样式
@mixin btnInlineBase() {
    cursor: pointer;

    background-color: $button_background_main;
    color: $button_text_color;
    text-align: center;

    display: inline-flex;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    box-sizing: border-box;
    border: 1px solid $button_border_main;

    white-space: nowrap;
    vertical-align: middle;
    -ms-touch-action: manipulation;
    touch-action: manipulation;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;

    // 按钮大少
    @include btnSize();

    // 点击反馈
    @include btnActive($buton_main_background_active, $button_border_main, 'default');
    // 次级按钮样式
    @include btnSubStyle();
    // 特殊按钮样式
    @include btnSpecialStyle();

    &#{$cls}-inactive {
        opacity: 0.3;
        background-color: $button_background_inactive;
    }

    &#{$cls}-disabled {
        opacity: 0.2;
        background-color: $button_background_disabled;
    }
}

//按钮class
#{$cls} {
    @include btnBase();
}

//按钮class
#{$cls}-inline {
    @include btnInlineBase();
}

// 按钮快
#{$cls-group} {
    display: flex;
    align-items: center;
    &#{$cls-group}-center{
        justify-content: space-between;
    }
    &#{$cls-group}-left{
        justify-content: flex-start;
    }
    &#{$cls-group}-right{
        justify-content: flex-end;
    }
    #{$cls} {
        flex: 1;
        &:not(:first-child) {
            margin-left: 12px;
        }
        &#{$cls}-inline {
            width: auto;
            flex: 0!important;
        }
    }
    
}