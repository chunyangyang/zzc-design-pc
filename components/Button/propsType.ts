export interface ButtonProps {
    prefixCls?: string,
    size?: string | Boolean,
    inline?: boolean,
    disabled?: boolean,
    inactive?: boolean,
    type?: string,
    onClick?: any,
    className?: string,
    style?: React.CSSProperties,
    activeStyle?: React.CSSProperties,
    activeClassName?: string
    htmlType?: string
}

export interface ButtonGroupProps {
    prefixCls?: string,
    className?: string,
    style?: React.CSSProperties,
    direction?: string
}