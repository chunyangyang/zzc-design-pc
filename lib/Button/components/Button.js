'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _TouchFeedback = require('../../TouchFeedback');

var _TouchFeedback2 = _interopRequireDefault(_TouchFeedback);

var _config = require('../../_util/config');

var _config2 = _interopRequireDefault(_config);

var _ButtonGroup = require('./ButtonGroup');

var _ButtonGroup2 = _interopRequireDefault(_ButtonGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = function (_PureComponent) {
    (0, _inherits3.default)(Button, _PureComponent);

    function Button(props) {
        (0, _classCallCheck3.default)(this, Button);
        return (0, _possibleConstructorReturn3.default)(this, (Button.__proto__ || (0, _getPrototypeOf2.default)(Button)).call(this, props));
    }

    (0, _createClass3.default)(Button, [{
        key: 'setActiveClassName',
        value: function setActiveClassName(activeClassName, type) {
            var _classNames;

            var activeClassNameIsNone = activeClassName === '';
            var className = (0, _classnames2.default)((_classNames = {}, (0, _defineProperty3.default)(_classNames, this.props.prefixCls + '-active-' + type, activeClassNameIsNone), (0, _defineProperty3.default)(_classNames, '' + activeClassName, !activeClassNameIsNone), _classNames));
            return className;
        }
    }, {
        key: 'render',
        value: function render() {
            var _classes;

            var _props = this.props,
                inactive = _props.inactive,
                children = _props.children,
                className = _props.className,
                prefixCls = _props.prefixCls,
                type = _props.type,
                size = _props.size,
                inline = _props.inline,
                disabled = _props.disabled,
                style = _props.style,
                activeStyle = _props.activeStyle,
                activeClassName = _props.activeClassName,
                onClick = _props.onClick,
                htmlType = _props.htmlType;

            var btnClassNames = '';
            ;
            var classes = (_classes = {}, (0, _defineProperty3.default)(_classes, prefixCls + '-' + size, size), (0, _defineProperty3.default)(_classes, prefixCls + '-' + type, type !== ''), (0, _defineProperty3.default)(_classes, prefixCls + '-disabled', disabled), (0, _defineProperty3.default)(_classes, prefixCls + '-inactive', inactive), _classes);
            if (inline) {
                btnClassNames = (0, _classnames2.default)(prefixCls, prefixCls + '-inline', className, classes);
            } else {
                btnClassNames = (0, _classnames2.default)(prefixCls, className, classes);
            }
            return _react2.default.createElement(
                _TouchFeedback2.default,
                { activeStyle: activeStyle ? activeStyle : {}, activeClassName: this.setActiveClassName(activeClassName, type), disabled: inactive || disabled },
                htmlType ? _react2.default.createElement(
                    'button',
                    { type: htmlType, className: btnClassNames, style: style, onClick: onClick },
                    children
                ) : _react2.default.createElement(
                    'div',
                    { className: btnClassNames, style: style, onClick: onClick },
                    children
                )
            );
        }
    }]);
    return Button;
}(_react.PureComponent);

exports.default = Button;

Button.defaultProps = {
    prefixCls: _config2.default.cls + '-button',
    size: false,
    inline: false,
    disabled: false,
    inactive: false,
    type: 'default',
    htmlType: null,
    onClick: function onClick() {},

    className: '',
    style: {},
    activeStyle: {},
    activeClassName: ''
};
Button.ButtonGroup = _ButtonGroup2.default;