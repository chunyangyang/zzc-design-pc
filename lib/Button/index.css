.zds-slide-bottom-enter {
  position: absolute;
  transform: translateY(100%);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.47, 0.65, 0.61, 0.9);
  animation-play-state: paused; }

.zds-slide-bottom-enter.zds-slide-bottom-enter-active {
  -webkit-animation-name: zdsSlideBottomEnter;
  animation-name: zdsSlideBottomEnter;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-slide-bottom-leave {
  position: absolute;
  transform: translateY(100%);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-play-state: paused; }

.zds-slide-bottom-leave.zds-slide-bottom-leave-active {
  -webkit-animation-name: zdsSlideBottomLeave;
  animation-name: zdsSlideBottomLeave;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsSlideBottomEnter {
  0% {
    transform: translateY(100%); }
  100% {
    transform: translateY(0); } }

@keyframes zdsSlideBottomLeave {
  0% {
    transform: translateY(0); }
  100% {
    transform: translateY(100%); } }

.zds-slide-top-enter {
  position: absolute;
  transform: translateY(-100%);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.47, 0.65, 0.61, 0.9);
  animation-play-state: paused; }

.zds-slide-top-enter.zds-slide-top-enter-active {
  -webkit-animation-name: zdsSlideTopEnter;
  animation-name: zdsSlideTopEnter;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-slide-top-leave {
  position: absolute;
  transform: translateY(0);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-play-state: paused; }

.zds-slide-top-leave.zds-slide-top-leave-active {
  -webkit-animation-name: zdsSlideTopLeave;
  animation-name: zdsSlideTopLeave;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsSlideTopEnter {
  0% {
    transform: translateY(-100%); }
  100% {
    transform: translateY(0); } }

@keyframes zdsSlideTopLeave {
  0% {
    transform: translateY(0); }
  100% {
    transform: translateY(-100%); } }

.zds-slide-left-enter {
  position: absolute;
  transform: translateX(-100%);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.47, 0.65, 0.61, 0.9);
  animation-play-state: paused; }

.zds-slide-left-enter.zds-slide-left-enter-active {
  -webkit-animation-name: zdsSlideLeftEnter;
  animation-name: zdsSlideLeftEnter;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-slide-left-leave {
  position: absolute;
  transform: translateX(0);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-play-state: paused; }

.zds-slide-left-leave.zds-slide-left-leave-active {
  -webkit-animation-name: zdsSlideLeftLeave;
  animation-name: zdsSlideLeftLeave;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsSlideLeftEnter {
  0% {
    transform: translateX(-100%); }
  100% {
    transform: translateX(0); } }

@keyframes zdsSlideLeftLeave {
  0% {
    transform: translateX(0); }
  100% {
    transform: translateX(-100%); } }

.zds-slide-right-enter {
  position: absolute;
  transform: translateX(100%);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.47, 0.65, 0.61, 0.9);
  animation-play-state: paused; }

.zds-slide-right-enter.zds-slide-right-enter-active {
  -webkit-animation-name: zdsSlideRightEnter;
  animation-name: zdsSlideRightEnter;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-slide-right-leave {
  position: absolute;
  transform: translateX(0);
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-play-state: paused; }

.zds-slide-right-leave.zds-slide-right-leave-active {
  -webkit-animation-name: zdsSlideRightLeave;
  animation-name: zdsSlideRightLeave;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsSlideRightEnter {
  0% {
    transform: translateX(100%); }
  100% {
    transform: translateX(0); } }

@keyframes zdsSlideRightLeave {
  0% {
    transform: translateX(0); }
  100% {
    transform: translateX(100%); } }

.zds-fade-enter {
  opacity: 0;
  animation-duration: .2s;
  animation-fill-mode: both;
  animation-play-state: paused; }

.zds-fade-enter.zds-fade-enter-active {
  -webkit-animation-name: zdsFadeIn;
  animation-name: zdsFadeIn;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-fade-leave {
  opacity: 0;
  animation-duration: .2s;
  animation-fill-mode: forwards;
  animation-play-state: paused; }

.zds-fade-leave.zds-fade-leave-active {
  -webkit-animation-name: zdsFadeOut;
  animation-name: zdsFadeOut;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsFadeIn {
  0% {
    opacity: 0; }
  100% {
    opacity: 1; } }

@keyframes zdsFadeOut {
  0% {
    opacity: 1; }
  100% {
    opacity: 0; } }

.zds-zoom-enter {
  opacity: 0;
  transform: scale(0.3);
  animation-duration: .3s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.68, -0.55, 0.265, 1.55);
  animation-play-state: paused; }

.zds-zoom-enter.zds-zoom-enter-active {
  -webkit-animation-name: zdsZoomIn;
  animation-name: zdsZoomIn;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

.zds-zoom-leave {
  animation-duration: .3s;
  animation-fill-mode: both;
  animation-timing-function: cubic-bezier(0.68, -0.55, 0.265, 1.55);
  animation-play-state: paused; }

.zds-zoom-leave.zds-zoom-leave-active {
  -webkit-animation-name: zdsZoomOut;
  animation-name: zdsZoomOut;
  -webkit-animation-play-state: running;
  animation-play-state: running; }

@keyframes zdsZoomIn {
  0% {
    opacity: 0;
    transform: scale(0.3); }
  100% {
    opacity: 1;
    transform: scale(1); } }

@keyframes zdsZoomOut {
  0% {
    opacity: 1;
    transform: scale(1); }
  100% {
    opacity: 0;
    transform: scale(0.3); } }

.zds-button {
  cursor: pointer;
  font-size: 17px;
  font-weight: 500;
  padding: 11.5px 0;
  line-height: 26px;
  background-color: #2871F7;
  color: #fff;
  text-align: center;
  border: 1px solid #2871F7;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none; }

.zds-button.zds-button-active-default {
  background: #1F58C0;
  border-color: #1F58C0; }

.zds-button.zds-button-sub {
  background-color: #fff;
  color: #2871F7;
  border: 1px solid #2871F7; }

.zds-button.zds-button-sub.zds-button-active-sub {
  background: rgba(40, 113, 247, 0.12);
  border-color: #2871F7; }

.zds-button.zds-button-sub.zds-button-disabled {
  opacity: 0.3;
  background-color: #fff;
  color: #696E7C;
  border: 1px solid #696E7C; }

.zds-button.zds-button-special {
  background-color: #fff;
  color: #696E7C;
  border: 1px solid #696E7C; }

.zds-button.zds-button-special.zds-button-active-special {
  background: #fff;
  border-color: rgba(105, 110, 124, 0.5); }

.zds-button.zds-button-inactive {
  opacity: 0.3;
  background-color: #276FF1; }

.zds-button.zds-button-disabled {
  opacity: 0.2;
  background-color: #696E7C;
  border: 1px solid #696E7C; }

.zds-button-inline {
  cursor: pointer;
  background-color: #2871F7;
  color: #fff;
  text-align: center;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  box-sizing: border-box;
  border: 1px solid #2871F7;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  font-weight: 500;
  padding: 7.5px 34.5px;
  line-height: 20px; }

.zds-button-inline.zds-button-middle {
  font-size: 13px;
  height: 30px;
  font-weight: 500;
  padding: 4.5px 18px;
  line-height: 20px; }

.zds-button-inline.zds-button-small {
  font-size: 13px;
  font-weight: 500;
  padding: 1px 18px;
  line-height: 20px; }

.zds-button-inline.zds-button-active-default {
  background: #1F58C0;
  border-color: #2871F7; }

.zds-button-inline.zds-button-sub {
  background-color: #fff;
  color: #2871F7;
  border: 1px solid #2871F7; }

.zds-button-inline.zds-button-sub.zds-button-active-sub {
  background: rgba(40, 113, 247, 0.12);
  border-color: #2871F7; }

.zds-button-inline.zds-button-sub.zds-button-disabled {
  opacity: 0.3;
  background-color: #fff;
  color: #696E7C;
  border: 1px solid #696E7C; }

.zds-button-inline.zds-button-special {
  background-color: #fff;
  color: #696E7C;
  border: 1px solid #696E7C; }

.zds-button-inline.zds-button-special.zds-button-active-special {
  background: #fff;
  border-color: rgba(105, 110, 124, 0.5); }

.zds-button-inline.zds-button-inactive {
  opacity: 0.3;
  background-color: #276FF1; }

.zds-button-inline.zds-button-disabled {
  opacity: 0.2;
  background-color: #696E7C; }

.zds-button-group {
  display: flex;
  align-items: center; }

.zds-button-group.zds-button-group-center {
  justify-content: space-between; }

.zds-button-group.zds-button-group-left {
  justify-content: flex-start; }

.zds-button-group.zds-button-group-right {
  justify-content: flex-end; }

.zds-button-group .zds-button {
  flex: 1; }

.zds-button-group .zds-button:not(:first-child) {
  margin-left: 12px; }

.zds-button-group .zds-button.zds-button-inline {
  width: auto;
  flex: 0 !important; }
