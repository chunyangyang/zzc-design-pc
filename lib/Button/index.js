'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Button = require('./components/Button');

var _Button2 = _interopRequireDefault(_Button);

require('../style/index');

require('./index.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _Button2.default;