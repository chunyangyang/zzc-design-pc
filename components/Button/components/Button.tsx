import React, { PureComponent } from 'react';
import classNames from 'classnames';
import TouchFeedback from '../../TouchFeedback';
import config from '../../_util/config';
import ButtonGroup from './ButtonGroup';
import { ButtonProps } from '../propsType';
export default class Button extends PureComponent<ButtonProps, any> {
    constructor(props) {        
        super(props);
    }
    static defaultProps = {
        prefixCls: `${config.cls}-button`,
        size: false,
        inline: false,
        disabled: false,
        inactive: false,
        type: 'default',
        htmlType: null,
        onClick() { },
        className: '',
        style: {
        },
        activeStyle: {
        },
        activeClassName: ''
    };

    static ButtonGroup = ButtonGroup;

    setActiveClassName(activeClassName?: string, type?: string): string {
        const activeClassNameIsNone: boolean = activeClassName === '';
        const className = classNames({
            [`${this.props.prefixCls}-active-${type}`]: activeClassNameIsNone,
            [`${activeClassName}`]: !activeClassNameIsNone
        });
        return className;
    }

    render() {                
        const { inactive, children, className, prefixCls, type, size, inline, disabled, style, activeStyle, activeClassName, onClick, htmlType } = this.props;
        let btnClassNames: string = ''; 
        interface classType { [propName: string]: any };
            let classes: classType = {
                [`${prefixCls}-${size}`]: size,
                [`${prefixCls}-${type}`]: type !== '',                
                [`${prefixCls}-disabled`]: disabled,
                [`${prefixCls}-inactive`]: inactive
            };       
        if (inline) {
            btnClassNames = classNames(
                prefixCls,
                `${prefixCls}-inline`,                
                className,
                classes
            );
        } else {
            btnClassNames = classNames(
                prefixCls,
                className,
                classes
            );                        
        }
        return (
            <TouchFeedback
                activeStyle={activeStyle ? activeStyle : {}}
                activeClassName={this.setActiveClassName(activeClassName, type)}
                disabled={inactive || disabled}
            >
                {
                    htmlType ?
                        <button
                            type={htmlType}
                            className={
                                btnClassNames
                            }
                            style={style}
                            onClick={onClick}
                        >
                            {children}
                        </button> :
                        <div
                            className={
                                btnClassNames
                            }
                            style={style}
                            onClick={onClick}
                        >
                            {children}
                        </div>
                }

            </TouchFeedback>
        );
    }
}