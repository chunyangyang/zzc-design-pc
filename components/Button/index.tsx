import Button from './components/Button';
import '../style/index';
import './index.scss';

export default Button;