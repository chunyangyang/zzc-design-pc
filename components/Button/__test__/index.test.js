import React from 'react';
import { mount, shallow, render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Button from '../index.tsx';

test( 'Button', () => {
  const wrapper = render( <div>
    <Button>defualt</Button>
    <Button type="main">main</Button>
    <Button type="main-lh">main-lh</Button>
    <Button type="main-dk">main-dk</Button>
    <Button type="sub">sub</Button>
    <Button type="sub-lh">sub-lh</Button>
    <Button type="sub-dk">sub-dk</Button>
    <Button disabled>disabled</Button>
    <Button type="main">default main</Button>
    <Button size='lg' type="main">lg main</Button>
    <Button size='sm' type="main">sm main</Button>
    <Button size='xs' type="main">xs main</Button>
    <Button inline size='sm'>inline1</Button>
    <Button inline size='sm' type="main">inline2</Button>
    <Button inline size='sm' type="sub">inline3</Button>
    <Button ghost inline size='xs' type="main">inline2</Button>
    <Button ghost inline size='xs' type="sub">inline3</Button>
    <Button full type="main">full main</Button>
    <Button activeStyle={{ background: 'blue' }} style={{ background: 'green', color: '#fff' }}>自定义样式</Button>
    <Button activeClassName='custom'>自定义active</Button>
    <Button activeStyle={false}>禁止active</Button>
  </div> );
  expect( renderToJson( wrapper ) ).toMatchSnapshot();
} );