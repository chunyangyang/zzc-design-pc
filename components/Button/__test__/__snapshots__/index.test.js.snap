// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<div>
  <div
    class="zds-button zds-button-default"
  >
    defualt
  </div>
  <div
    class="zds-button zds-button-main"
  >
    main
  </div>
  <div
    class="zds-button zds-button-main-lh"
  >
    main-lh
  </div>
  <div
    class="zds-button zds-button-main-dk"
  >
    main-dk
  </div>
  <div
    class="zds-button zds-button-sub"
  >
    sub
  </div>
  <div
    class="zds-button zds-button-sub-lh"
  >
    sub-lh
  </div>
  <div
    class="zds-button zds-button-sub-dk"
  >
    sub-dk
  </div>
  <div
    class="zds-button zds-button-default zds-button-disabled"
  >
    disabled
  </div>
  <div
    class="zds-button zds-button-main"
  >
    default main
  </div>
  <div
    class="zds-button zds-button-lg zds-button-main"
  >
    lg main
  </div>
  <div
    class="zds-button zds-button-sm zds-button-main"
  >
    sm main
  </div>
  <div
    class="zds-button zds-button-xs zds-button-main"
  >
    xs main
  </div>
  <div
    class="zds-button zds-button-sm zds-button-default zds-button-inline"
  >
    inline1
  </div>
  <div
    class="zds-button zds-button-sm zds-button-main zds-button-inline"
  >
    inline2
  </div>
  <div
    class="zds-button zds-button-sm zds-button-sub zds-button-inline"
  >
    inline3
  </div>
  <div
    class="zds-button zds-button-xs zds-button-main zds-button-inline zds-button-ghost"
  >
    inline2
  </div>
  <div
    class="zds-button zds-button-xs zds-button-sub zds-button-inline zds-button-ghost"
  >
    inline3
  </div>
  <div
    class="zds-button zds-button-full zds-button-main"
  >
    full main
  </div>
  <div
    class="zds-button zds-button-default"
    style="background:green;color:#fff"
  >
    自定义样式
  </div>
  <div
    class="zds-button zds-button-default"
  >
    自定义active
  </div>
  <div
    class="zds-button zds-button-default"
  >
    禁止active
  </div>
</div>
`;
